### PR TITLE
Add option for non-existent params

### PR DESCRIPTION
**Pull request by avgAGB**

### CODE REVIEW

File: warehouse/query-core/src/test/java/datawave/query/MiscQueryTest.java [hunk 1]
```diff
@@ -96,6 +97,57 @@ public void testEventThreshold() throws Exception {
         runTest(query, expect);
     }
 
+    @Test(expected = InvalidQueryException.class)
+    public void testFieldIgnoreParam1() throws Exception {
+        log.info("------  testEventThreshold  ------");
```
Comment (matthpeterson) [inline warehouse/query-core/src/test/java/datawave/query/MiscQueryTest.java]: nit: log message doesn't match test name

File: warehouse/query-core/src/test/java/datawave/query/MiscQueryTest.java [hunk 2]
```diff
@@ -96,6 +97,57 @@ public void testEventThreshold() throws Exception {
...
+        // setting event per day does not alter results
+        this.logic.setEventPerDayThreshold(1);
+        String phrase = RE_OP + "'.*a' && STATE == 'sta'";
+        String query = Constants.ANY_FIELD + phrase + "&& STATE == 'sta'";
+        String expect = this.dataManager.convertAnyField(phrase);
+
+        Map<String,String> options = new HashMap<>();
+
+        // this should allow the query to run successfully.
```
Comment (matthpeterson) [inline warehouse/query-core/src/test/java/datawave/query/MiscQueryTest.java]: This may be a copy and paste error on this comment

**Review by matthpeterson — DISMISSED** [review]

File: warehouse/query-core/src/test/java/datawave/query/MiscQueryTest.java (hunk 1 above)
Comment (avgAGB) [reply warehouse/query-core/src/test/java/datawave/query/MiscQueryTest.java]: Thanks, updated

File: warehouse/query-core/src/test/java/datawave/query/MiscQueryTest.java (hunk 2 above)
Comment (avgAGB) [reply warehouse/query-core/src/test/java/datawave/query/MiscQueryTest.java]: Will remove from the latest commit

**Review by mineralntl — DISMISSED** [review]

**Review by matthpeterson — DISMISSED** [review]

**Review by jjonesy16 — DISMISSED** [review]
Looks good to me, nice work!

File: warehouse/query-core/src/main/java/datawave/query/config/ShardQueryConfiguration.java [hunk 3]
```diff
@@ -1648,6 +1650,14 @@ public void setDateIndexTimeTravel(boolean dateIndexTimeTravel) {
         this.dateIndexTimeTravel = dateIndexTimeTravel;
     }
 
+    public boolean getIgnoreExistentFields() {
+        return ignoreNonExistentFields;
+    }
+
+    public void setIgnoreExistentFields(boolean ignoreNonExistentFields) {
```
Comment (awrigh11) [inline warehouse/query-core/src/main/java/datawave/query/config/ShardQueryConfiguration.java]: It looks like you forgot the "Non" in this method name and the one above.

Comment (avgAGB) [reply warehouse/query-core/src/main/java/datawave/query/config/ShardQueryConfiguration.java]: ah, good catch. I'll fix this and recommit

**Review by awrigh11 — APPROVED** [review]

**Review by jjonesy16 — APPROVED** [review]

**Pull request merged**